snapcraft/internal/lxd.py

```diff
@@ -137,7 +142,8 @@ def _setup_project(self):
                            os.path.basename(tar_filename))
         self._container_run(['mkdir', self._project_folder])
         self._push_file(tar_filename, dst)
-        self._container_run(['tar', 'xvf', os.path.basename(tar_filename)])
+        self._container_run(['tar', 'xvf', os.path.basename(tar_filename)],
+                            cwd=self._project_folder)
```

Comment:
I learned that in the upcoming lxc version, it's possible to just push the directory.
Not very useful though, as we have to support the version in trusty. Just fyi.